src/interpreter.rs

```diff
@@ -18,6 +19,95 @@ impl Execute for Expression {
+        let mut top: usize = 0;
+        let mut operand_stack: [OperandItem; 2] = [OperandItem::Val(false); 2];
+
+        for code in &self.codes {
+            match code {
+                LirCode::LogicalOperator(op) => {
+                    match op {
+                        LirLogicalOperators::And => {
+                            let left = evaluate_operand_item(operand_stack[top - 2], ctx, m);
+                            if !left {
+                                // short circuit
+                                // stack pop
+                                top -= 2;
+
+                                //stack push
+                                operand_stack[top] = OperandItem::Val(false);
+                                top += 1;
+                            } else {
+                                let right = evaluate_operand_item(operand_stack[top - 1], ctx, m);
+
+                                //stack pop
+                                top -= 2;
+                                //stack push
+                                operand_stack[top] = OperandItem::Val(right);
+                                top += 1;
+                            }
+                        }
+                        LirLogicalOperators::Or => {
+                            let left = evaluate_operand_item(operand_stack[top - 2], ctx, m);
+                            if left {
+                                // short circuit
+                                // stack pop
+                                top -= 2;
+
+                                //stack push
+                                operand_stack[top] = OperandItem::Val(true);
+                                top += 1;
+                            } else {
+                                let right = evaluate_operand_item(operand_stack[top - 1], ctx, m);
+
+                                //stack pop
+                                top -= 2;
+                                //stack push
+                                operand_stack[top] = OperandItem::Val(right);
+                                top += 1;
+                            }
+                        }
+                        LirLogicalOperators::Not => {
+                            //stack pop
+                            top -= 1;
+                            let operand = evaluate_operand_item(operand_stack[top], ctx, m);
+
+                            //stack push
+                            operand_stack[top] = OperandItem::Val(!operand);
+                            top += 1;
+                        }
+                    }
+                }
+                LirCode::Predicate(p) => {
+                    // push stack
+                    operand_stack[top] = OperandItem::Predicate(p);
+                    top += 1;
+                }
+            }
+        }
+        //stack pop
+        top -= 1;
+        if top != 0 {
+            panic!("There may be something wrong for the AST conversion!")
```

Comment:
Hint of validation: https://github.com/Kong/atc-router/blob/3f8f648b344d783e076b0f197cdd93102bd3d535/src/semantics.rs#L65